### PR TITLE
chore(notifications): narrow the scope to messaging

### PR DESCRIPTION
**Pull request by mareklibra**
## Hey, I just made a Pull Request!
Updating Notifications BEP to narrow the scope to a messaging system only.
Clarified some of the fields in the API.

Based on the discussion on Discord.

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: beps/0001-notifications-system/README.md [hunk 1]
```diff
@@ -96,15 +96,15 @@ The notification backend stores notification using the [database service](https:
...
 - Saved status
 - Creation date
 - Updated date (optional, for scoped notifications)
 - Payload:
   - Title
   - Description (optional)
   - Origin
-  - Link
+  - Link (optional)
+  - Additional links (optional)
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:107]: 👍

File: beps/0001-notifications-system/README.md [hunk 2]
```diff
@@ -113,6 +113,10 @@ The notification backend stores notification using the [database service](https:
 
 The recipients is **not** a list of users, but rather a filter that describes who should receive the notification. It must be possible to evaluate this filter in a database query, so that we can efficiently fetch all notifications for a given user. The same filter will also be used by the signal backend to determine which users should receive a signal.
 
+The read date is a timestamp of marking the notifications as read by the user. If missing, the notification is still unread.
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:116]: 👍

File: beps/0001-notifications-system/README.md [hunk 3]
```diff
@@ -113,6 +113,10 @@ The notification backend stores notification using the [database service](https:
...
+
+The saved status is a boolean indicating whether the user marked the notification for better visibility in the future (to "pin" the message).
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:118]: I think it makes sense to apply the design principle of "timestamps over booleans" here too. I.e. just like the read state, the saved state is also an optional timestamp. Although tbh it's not a super strong opinion that we need to use a timestamp here since it's arguably of little value since we already have the creation/update timestamps to sort by. Although it would allow us to highlight notifications that have been update since they were saved.

My stronger opinion here is that plain booleans are generally unnecessary and you're able to convey more information or allow for more evolvability with other data types. If we feel that the timestamp really is unnecessary then I'd instead think we could make this a saved state enum or object.

File: beps/0001-notifications-system/README.md [hunk 4]
```diff
@@ -132,6 +136,7 @@ If a scope is provided for a notification, it will either:
 - Update existing notification in that scope and origin so that it's unread considered as a new notification
 
 The idea of scope is to be able to reuse notifications about same information for example specific failing CI build.
+Payload of notifications without scope is meant to be immutable.
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:139]: This is unrelated to this PR, just bringing it up to encourage some thing around it: I don't really feel that "scope" name conveys the intent and behavior of this field, and it would be best to rename it to something else

File: beps/0001-notifications-system/README.md [hunk 5]
```diff
@@ -143,6 +148,8 @@ The link is a relative or absolute URL. As an example, it can be used:
 - by an external system to request an action within an asynchronous task
 - by a BE plugin to provide link to other part of the Backstage UI (i.e. to the Catalog)
 
+The additional links are an array of title-URL pairs. They can represent immediate actions on the notification (i.e. yes-no) or lead the user to additional details.
```
Comment (Rugvip) [inline beps/0001-notifications-system/README.md:151]: Keeping an 👀 on this. Feeling it's an alright direction for now, but we'll want to find a good balance between baseline data format and custom payload format. Primary purpose there being the ability to process notifications in the backend in a generic way. In the frontend we could use a custom payload and renderer, and even custom conversions to `WebNotifications`

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 👍 

Just a nit on the saved status format from me, but not a hill I'm gonna die on 😅

File: beps/0001-notifications-system/README.md (hunk 4 above)
Comment (mareklibra) [reply beps/0001-notifications-system/README.md:139]: Honestly, I would remove it completely. The use-case with CI can be handled via filtering "the latest of a topic" message.

File: beps/0001-notifications-system/README.md (hunk 3 above)
Comment (mareklibra) [reply beps/0001-notifications-system/README.md:118]: SUre, let's consistently use the Date.

File: beps/0001-notifications-system/README.md (hunk 4 above)
Comment (drodil) [reply beps/0001-notifications-system/README.md:139]: I wouldn't remove it. It's very many times use case that you want to re-use the same notification. For example in my Q&A plugin user creates a question. 2 other users replies to it so I don't want to send 2 notifications about having new replies for the same question.

Comment (drodil) [reply beps/0001-notifications-system/README.md:139]: What comes to naming, I really can't think better one 😆 If you have ideas, please comment!

Comment (mareklibra) [reply beps/0001-notifications-system/README.md:139]: I am not eager to remove it.

Anyway, the `scope` seems to be kind of `key` which is used to repeatedly find&edit the same message.
Technically, the recent `id` should be enough.

Comment (drodil) [reply beps/0001-notifications-system/README.md:139]: I see use cases for this. It's optional anyway so is there some harm in keeping it?

Comment (Rugvip) [reply beps/0001-notifications-system/README.md:139]: The benefit of having something like scope over ID is that you can avoid the need for plugins to store state about notifications, as they can drive the scope from their own state

File: beps/0001-notifications-system/README.md (hunk 3 above)
Comment (Rugvip) [reply beps/0001-notifications-system/README.md:118]: Could you update the text to reflect that?

File: beps/0001-notifications-system/README.md (hunk 5 above)
Comment (mareklibra) [reply beps/0001-notifications-system/README.md:151]: Removing the line, keeping the scope as it is.

File: beps/0001-notifications-system/README.md (hunk 3 above)
Comment (mareklibra) [reply beps/0001-notifications-system/README.md:118]: done
